lib/geminabox/server.rb

```diff
@@ -195,6 +195,10 @@ def handle_incoming_gem(gem)
         GemStore.create(gem, params[:overwrite])
       rescue GemStoreError => error
         error_response error.code, error.reason
+      else
+        if Geminabox.on_gem_received
+          Geminabox.on_gem_received.call(gem)
+        end
```

Comment:
We should think about some specifications:

1. return error_response if the callback errors
    * we can tell as the hook writer is responsible to the error handling, but I think it is safer to take care in geminabox codes
2. If the callback takes long time, the web server is occupied by the process. I think we have two options:
    * introduce some asynchronous library, and execute the callback asynchronously
    * Or, we can tell as the hook writer is responsible to the asynchronous execution.

Reply:
Sounds reasonable to me. Could you write about that on README and fix codes (rescuing error silently)?